computerGroups.go

```diff
@@ -0,0 +1,122 @@
+package jamf
+
+import "fmt"
+
+const uriComputerGroups = "/JSSResource/computergroups"
+
+type ComputerGroupsResponse struct {
+	Count   int                         `xml:"size"`
+	Results []ComputerGroupListResponse `xml:"computer_group"`
+}
+
+type ComputerGroupListResponse struct {
+	ID      int    `xml:"id,omitempty"`
+	Name    string `xml:"name,omitempty"`
+	IsSmart bool   `xml:"is_smart,omitempty"`
+}
+
+type ComputerGroup struct {
+	ID            int                      `xml:"id"`
+	Name          string                   `xml:"name"`
+	IsSmart       bool                     `xml:"is_smart"`
+	Site          Site                     `xml:"site"`
+	Criteria      []ComputerGroupCriterion `xml:"criteria>criterion"`
+	CriteriaCount int                      `xml:"criteria>size"`
+	Computers     []ComputerGroupResp      `xml:"computers>computer"`
+	ComputerCount int                      `xml:"computers>size"`
+}
+
+type ComputerGroupRequest struct {
+	Name      string                      `xml:"name"`
+	IsSmart   bool                        `xml:"is_smart"`
+	Site      Site                        `xml:"site"`
+	Criteria  []ComputerGroupCriterion    `xml:"criteria>criterion"`
+	Computers []ComputerGroupListResponse `xml:"computers>computer,omitempty"`
+}
+
+type ComputerGroupCriterion struct {
+	Name         string             `xml:"name"`
+	Priority     int                `xml:"priority"`
+	AndOr        ComputerGroupAndOr `xml:"and_or"`
+	SearchType   string             `xml:"search_type"`
+	SearchValue  string             `xml:"value"`
+	OpeningParen bool               `xml:"opening_paren"`
+	ClosingParen bool               `xml:"closing_paren"`
+}
+
+type ComputerGroupAndOr string
+
+const (
+	And ComputerGroupAndOr = "and"
+	Or  ComputerGroupAndOr = "or"
+)
+
+func (c *Client) GetComputerGroupIdByName(name string) (int, error) {
+	var id int
+	d, err := c.GetComputerGroups()
+	if err != nil {
+		return -1, err
```

Comment:
[IMO] `return int, err`

This reason is it returns an error, so initial value of int  is fine.